test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs

```diff
@@ -202,5 +203,68 @@ public void DthCompilation_RestoreComplete_OnEmptyConsoleApp(string flavor, stri
                 Assert.NotNull(after.Payload["Dependencies"]["System.Console"]);
             }
         }
+
+        public static IEnumerable<object[]> UnresolvedDependencyTestData
+        {
+            get
+            {
+                foreach (var combination in RuntimeComponents)
+                {
+                    yield return combination.Concat(new object[] { 1, "UnresolvedProjectSample", "EmptyLibrary", "Unresolved" }).ToArray();
+
+                    yield return combination.Concat(new object[] { 1, "UnresolvedPackageSample", "NoSuchPackage", "Unresolved" }).ToArray();
+
+                    yield return combination.Concat(new object[] { 2, "UnresolvedProjectSample", "EmptyLibrary", "Unresolved" }).ToArray();
+
+                    yield return combination.Concat(new object[] { 2, "UnresolvedPackageSample", "NoSuchPackage", "Unresolved" }).ToArray();
+
+                    yield return combination.Concat(new object[] { 3, "UnresolvedProjectSample", "EmptyLibrary", "Project" }).ToArray();
+
+                    // Unresolved package dependency's type is still Unresolved
+                    yield return combination.Concat(new object[] { 3, "UnresolvedPackageSample", "NoSuchPackage", "Unresolved" }).ToArray();
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(UnresolvedDependencyTestData))]
+        public void DthCompilation_Initialize_UnresolvedDependency(
+            string flavor, string os, string architecture, int protocolVersion,
+            string testProjectName, string expectedUnresolvedDependency, string expectedUnresolvedType)
+        {
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+            var testProject = _fixture.GetTestProjectPath(testProjectName);
+
+            using (var server = DthTestServer.Create(runtimeHomePath, testProject))
+            using (var client = new DthTestClient(server, 0))
+            {
+                client.Initialize(testProject, protocolVersion);
+
+                var message = client.DrainTillFirst("Dependencies");
+                message.EnsureSource(server, client);
+
+                var dependencies = message.Payload["Dependencies"];
```

Comment:
Do you have an issue to track it? :) Its easier to do it now, but its up to you of course (btw it would caught another issue with wrapped project path that I sent you earlier)